bin/lib/backup-restore.include.sh

```diff
@@ -0,0 +1,52 @@
+# This file includes functions that are intended to be sourced by the do_backup_restore function (in bin/birdhouse)
+# It assumes that various environment variables are set including:
+#  - BIRDHOUSE_EXE
+#  - BIRDHOUSE_BACKUP_VOLUME
+#  - BIRDHOUSE_RESTORE_SNAPSHOT
+#  - BIRDHOUSE_BACKUP_DRY_RUN
+
+eval "$(${BIRDHOUSE_EXE} configs --print-log-command)"
+
+do_backup() {
+  description="$1"
+  command="$2"
+  stop_containers="$3"
+  [ -z "${description}" ] && log WARN "No description provided for this backup job. A description is strongly recommended."
+  [ -z "${command}" ] && log ERROR "No command provided for the backup job with description: '${description}'. A command is required. Skipping this backup." && return 1
+  [ "${BIRDHOUSE_BACKUP_DRY_RUN}" = 'true' ] && echo "${description}" && return 0
+  if [ -n "${stop_containers}" ]; then
+    log INFO "Stopping containers: ${stop_containers}"
+    ${BIRDHOUSE_EXE} --quiet compose stop ${stop_containers}
+  fi
+  log INFO "Backing up ${description} to the ${BIRDHOUSE_BACKUP_VOLUME} volume"
+  ${command} && log INFO "Backup of ${description} complete" || log ERROR "Backup of ${description} failed"
```

Comment:
I think the missing `description` should be logged and return an error.
The command needs to do `do_backup "" ...` explicitly anyway, so omitting a very basic description about what is invoked will simply create weird and impractical logs.